### DISCUSSION TITLE
How can I disable TailwindCSS for a specific element?

### DISCUSSION THREAD

**Discussion by albingroen** [tailwindlabs/tailwindcss#5981] [Help]
I have case where I want to render html emails in my application, but I don't want TailwindCSS to affect the styling of these emails.

Is there any way I could disable TailwindCSS for just that element? I have tried rendering the emails inside of iframes, which do solve the css problem, but comes with other issues.

Help appreciated 🙏

**Comment by simonswiss** [comment]
Hey!

Mid-way through reading the question I was thinking `iframe`, but as you mention, there are other issues that can come with those.

Tailwind CSS doesn't have a built-in feature to "opt-out" of the `preflight` styles for just an element, unfortunately.

**Reply by albingroen** [reply]
Okay, then I understand. I went with iframes for now. It would be cool though if you could just add like a `tw-none` class or something to disable Tailwind styles for that element and all its children.

**Reply by BrammyS** [reply]
Adding something like this would be great!

**Reply by Abhishek-stockdaddy** [reply]
yes we really need it

**Comment by arthurdenner** [comment]
Today I learned about the `all` CSS property ([MDN page](https://developer.mozilla.org/en-US/docs/Web/CSS/all)).

> The all shorthand property resets all of an element's properties except unicode-bidi, direction and CSS Custom Properties.
> It can set properties to their initial or inherited values, or to the values specified in another stylesheet origin.

I think it can solve some of your needs, especially `all: initial`.
I've created a small plugin for it: https://github.com/arthurdenner/tailwindcss-all
Demo: https://tailwindcss-all.netlify.app

<details>
<summary>Screenshots</summary>

![comparison screenshot](https://user-images.githubusercontent.com/13774309/157909181-ed128564-f0de-4f57-b520-347424f90944.png)
</details>

**Reply by Abhishek-stockdaddy** [reply]
even this is not working
i'm using it like this

`   <div className="all-initial">
         <div className="text-center bg-gray-400 p-40 all-initial">About</div>
    </div>`

**Reply by Abhishek-stockdaddy** [reply]
okay nvm it's working

**Reply by Bedrock02** [reply]
wow thank you for this

**Reply by modrictin** [reply]
WOW! THank you so much on this one!!!

**Reply by ralpmancawan** [reply]
THANK YOU!!!

**Comment by albingroen** [comment]
Just wanted to check in on this again. Has there been any work on this?

**Comment by jasminediaconu** [comment]
It would be of great help if something like this could actually be implemented in tailwind directly. 
I also have some components where I need the preflight disabled and overwriting the tailwind stylesheet is not handy.

**Comment by Livia89** [comment]
Hi! [👋](https://emojipedia.org/waving-hand/) Could be one solution: "pointer-events-none"

**Reply by albingroen** [reply]
> Hi! [👋](https://emojipedia.org/waving-hand/) Could be one solution: "pointer-events-none"

No this doesn't solve the issue at hand unfortunately

**Comment by ItzAmirreza** [comment]
Bump? Tried all: initial but still doesn't work :) Really need something to disable preflight for a specific element.

**Comment by Krak86** [comment]
It's possible with tailwind new [feature](https://tailwindcss.com/docs/typography-plugin) - just add `prose` class and it ignores tailwind classes

**Reply by IAPBenediktZwoelfer** [reply]
Would you tell us how haha

**Reply by Krak86** [reply]
Something like that:
```
<article class="prose">
  <h1>Garlic bread with cheese: What the science tells us</h1>
  <p>
    For years parents have espoused the health benefits of eating garlic bread with cheese to their
    children, with the food earning such an iconic status in our culture that kids will often dress
    up as warm, cheesy loaf for Halloween.
  </p>
  <p>
    But a recent study shows that the celebrated appetizer may be linked to a series of rabies cases
    springing up around the country.
  </p>
  <!-- ... -->
</article>
```
And add appropriate plugin to tailwind config:

```
/** @type {import('tailwindcss').Config} */
module.exports = {
  theme: {
    // ...
  },
  plugins: [
    require('@tailwindcss/typography'),
    // ...
  ],
}
```

**Reply by lsmolic** [reply]
Saved me a bunch. Thank you!

**Comment by wwangwe** [comment]
You can use the official _**tailwindcss/typography**_ plugin that @Krak86 is explaining above. It does exactly what you're looking for and even exceeds expectations.

`npm install -D @tailwindcss/typography`
```
/** @type {import('tailwindcss').Config} */
module.exports = {
  theme: {
    // ...
  },
  plugins: [
    require('@tailwindcss/typography'),
    // ...
  ],
}
```
Now you can use the `prose` classes to add sensible typography styles to any vanilla HTML

See the [documentation](https://tailwindcss.com/docs/typography-plugin) for more details.

**Reply by OluAgunloye** [reply]
Ran into this issue myself, glad there's a fix now.
EDIT: Actually this doesn't work to eliminate the pre-flight check specifically for some elements, which is the desired effect.

**Comment by Sidmaz666** [comment]
This is the Solution that I came up with:

1. Add the Following CSS to `tailwind.css` file.
```
 .revert-tailwind{
  all: initial;
}

.revert-tailwind > * {
  all: revert;
}
```
2. Compile Tailwind CSS to the `main.css` stylesheet and use the class as follow:

```
<div class="bg-black text-white text-xl">
Affected By Tailwind
<div class="revert-tailwind">
<h1>Uneffected By Tailwind!</h1>
</div>
</div>
```

**Reply by aligajani** [reply]
How can I make this work with input element outline color?

**Reply by sovetski** [reply]
Looks very good but did not worked for me... Tailwind base styling overrides these changes, I tried this but it did not work too (in SCSS)

```scss
.revert-tailwind {
	all: initial !important;
	font-size: initial !important;
	font-weight: initial !important;

	> * {
		all: initial !important;
		font-size: initial !important;
		font-weight: initial !important;
	}
}
```

**Reply by Sidmaz666** [reply]
> Looks very good but did not worked for me... Tailwind base styling overrides these changes, I tried this but it did not work too (in SCSS)
> 
> ```css-scss
> .revert-tailwind {
> 	all: initial !important;
> 	font-size: initial !important;
> 	font-weight: initial !important;
> 
> 	> * {
> 		all: initial !important;
> 		font-size: initial !important;
> 		font-weight: initial !important;
> 	}
> }
> ```

Well, your SCSS doesn't look like the solution provided by me the correct SCSS will be:


```
revert-tailwind {
  all: initial;
 /*font-size: initial*/
  > * {
    all: revert;
    /* font-size: revert */
  }
}
```

You can add `!important` . This should work, and use the revert-tailwind class in a parent it's child should take the effects.

**Reply by Sidmaz666** [reply]
> How can I make this work with input element outline color?

Wrap the input element with a parent div with class revert-tailwind, use style attribute to style the input element

**Reply by MwanPygmay** [reply]
This solution does not seem to work well with borders:
Basic html result:
![image](https://github.com/tailwindlabs/tailwindcss/assets/59645813/24f0077a-c151-4cdf-ae8c-138b3b842991)
Result in my  app, using the revert-tailwind class on parent div:
![image](https://github.com/tailwindlabs/tailwindcss/assets/59645813/636b9a46-57f6-4924-a6c6-5493a8e799aa)

**Comment by sovetski** [comment]
Any news on this subject? In my case, I am using TinyMCE to manage my website content.

When I edit the content I see the browser's basic styling, but when I go on the preview page, as tailwind base is included on all website it also impact my content so I am getting a completely different result between Edit and Preview.

Something like @Sidmaz666 suggested, `.tailwind-ignore` etc. will be very perfect

**Reply by Sidmaz666** [reply]
> Any news on this subject? In my case, I am using TinyMCE to manage my website content.
> 
> When I edit the content I see the browser's basic styling, but when I go on the preview page, as tailwind base is included on all website it also impact my content so I am getting a completely different result between Edit and Preview.
> 
> Something like @Sidmaz666 suggested, `.tailwind-ignore` etc. will be very perfect

Yeah sure, just rename the `.revert-tailwind` class to `.tailwind-ignore`

**Comment by yabdab** [comment]
This worked for me ...

https://www.npmjs.com/package/tailwindcss-scoped-preflight

I used the `remove:` parameter to exclude certain tags from preflight.

Hoping this helps someone else 😉

**Reply by MwanPygmay** [reply]
works great indeed!

**Reply by martinpa13py** [reply]
this works really well

**Comment by SaahBrice** [comment]
something like all: revert-layer;
or to be more specific, color:revert-layer;
